datafusion/functions/src/datetime/common.rs

```diff
@@ -227,46 +229,34 @@ where
         // if the first argument is a scalar utf8 all arguments are expected to be scalar utf8
         ColumnarValue::Scalar(scalar) => match scalar {
             ScalarValue::Utf8(a) | ScalarValue::LargeUtf8(a) => {
-                let mut val: Option<Result<ColumnarValue>> = None;
-                let mut err: Option<DataFusionError> = None;
+                let a = a.as_ref();
+                // ASK: Why do we trust `a` to be non-null at this point?
```

Comment:
I don't think we do trust a to be non-null -- the current code on main has an explicitly `match` that handles `None` doesn't it? Maybe I am misreading the diff

```rust
                    None => (),
```

🤔 

However, clearly we don't have test coverage for this

Reply:
Previously, in this code path, if `a` was None, then the `err` that we unwrap later down in that same block would also be None, causing a more confusing error. So since we're comfortable unwrapping the err later on, we must be confident `a` is `Some(_)`.